zfsbootmenu/lib/zfsbootmenu-core.sh

```diff
@@ -1667,9 +1667,16 @@ load_key() {
     return 0
   fi
 
-  # Default to 0 when unset
-  [ -n "${CLEAR_SCREEN}" ] || CLEAR_SCREEN=0
```

Comment:
What wasn't working with this logic? It seemed to be fine as-is - there's more complexity with the new checks - is that solving a specific problem?

Reply:
No, this is just conflating a couple of things. The only thing that matters for the key caching problem is limiting the scope of the mutex to the current PID.

The changes to CLEAR_SCREEN and NO_CACHE were just an attempt to normalize the value and also avoid potential (but nonexistent) issues with numeric comparisons.